### PR TITLE
TRUNK-5033 public static fields should be final

### PR DESCRIPTION
**Pull request by abarne**
## Description of what I changed

Changed public static variables to final in OpenmrsConstants.java, SchedulerConstants.java, and security.java. KEY_OPENMRS_APPLICATION_DATA_DIRECTORY in OpenmrsConstants.java is accessed in other parts of the code so I added get and set methods.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5033

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can you fix the ticket url on this pull request?

**Comment by abarne** [comment]
Fixed the ticket URL. Didn't notice I left out the number.

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java [hunk 1]
```diff
@@ -42,7 +42,16 @@
 	
 	private static final Logger log = LoggerFactory.getLogger(OpenmrsConstants.class);
 	
-	public static String KEY_OPENMRS_APPLICATION_DATA_DIRECTORY = "OPENMRS_APPLICATION_DATA_DIRECTORY";
+	
+	private static String KEY_OPENMRS_APPLICATION_DATA_DIRECTORY = "OPENMRS_APPLICATION_DATA_DIRECTORY";
+	
+	public static void SET_KEY_OPENMRS_APPLICATION_DATA_DIRECTORY(String name){
+		KEY_OPENMRS_APPLICATION_DATA_DIRECTORY = name;
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: Is this supposed to be a method? Did you take a look at our conventions at https://wiki.openmrs.org/display/docs/Java+Conventions
Was this change even part of the ticket?

Comment (abarne) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: The change was not specifically stated in the ticket. But by changing KEY_OPENMRS_APPLICATION_DATA_DIRECTORY to a public static final blocked access to other parts of the code, resulting in a failed compile. So I changed it to private static and added the get and set methods. I will take a look at the conventions to see what I need to fix.

Comment (abarne) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: Fixed my code to adhere to the coding conventions. Would you like me to push a commit to this?

**Comment by teleivo** [comment]
@abarne please update your PR title + description to our convention. see for example this one https://github.com/openmrs/openmrs-core/pull/2107

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java [hunk 2]
```diff
@@ -42,13 +42,36 @@
 	
 	private static final Logger log = LoggerFactory.getLogger(OpenmrsConstants.class);
 	
-	public static String KEY_OPENMRS_APPLICATION_DATA_DIRECTORY = "OPENMRS_APPLICATION_DATA_DIRECTORY";
+	
+	private static String KEY_OPENMRS_APPLICATION_DATA_DIRECTORY = "OPENMRS_APPLICATION_DATA_DIRECTORY";
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: Forgot to add final here

**Comment by dkayiwa** [comment]
@abarne are you still working on this?

**Comment by dkayiwa** [comment]
@abarne ping me to reopen this whenever you are ready to resume. 😊

**Pull request closed without merging**